src/selectors/operators/numeric/BinaryNumericOperator.js

```diff
@@ -4,21 +4,24 @@ import Sequence from '../../dataTypes/Sequence';
 import Selector from '../../Selector';
 import createAtomicValue from '../../dataTypes/createAtomicValue';
 
+import YearMonthDuration from '../../dataTypes/valueTypes/YearMonthDuration';
+import DayTimeDuration from '../../dataTypes/valueTypes/DayTimeDuration';
+
 function executeOperator (kind, a, b) {
-    switch (kind) {
-        case '+':
-            return a + b;
-        case '-':
-            return a - b;
-        case '*':
-            return a * b;
-        case 'div':
-            return a / b;
-        case 'idiv':
-            return Math.trunc(a / b);
-        case 'mod':
-            return a % b;
-    }
+	switch (kind) {
```

Comment:
Alternatively, this seems very similar to calling functions. The approach taken there (data-driven type checks + dispatch) could possibly work here, instead of the massive if / switch blocks.